tests/functional/test_utils.py

```diff
@@ -40,3 +44,24 @@ def test_can_load_token(self):
         loader = FileWebIdentityTokenLoader(self.token_file)
         token = loader()
         self.assertEqual(self.token, token)
+
+
+class TestInstanceMetadataFetcher(unittest.TestCase):
+    def test_catch_retryable_http_errors(self):
+        with mock.patch('botocore.httpsession.URLLib3Session.send') as send_mock:
+            fetcher = InstanceMetadataFetcher()
+            send_mock.side_effect = EndpointConnectionError(endpoint_url="foo")
+            fetcher.retrieve_iam_role_credentials()
+        self.assertEquals(send_mock.call_count, 2)
+        self.assertEquals(
+            urlparse(send_mock.call_args_list[0][0][0].url).netloc,
```

Comment:
Can we use `get_base_url` from the fetcher to assert this is actually the IMDS url we're calling?